plugins/woocommerce/src/Admin/Features/ProductBlockEditor/Init.php

```diff
@@ -210,35 +218,137 @@ public function set_current_screen_to_block_editor_if_wc_admin() {
+				'product_template_product_data' => array(
+					'type' => 'grouped',
+				),
+			),
+			'external' => array(
+				'product_template_id'           => 'affiliate-product-template',
+				'product_template_title'        => __( 'Affiliate product', 'woocommerce' ),
+				'product_template_description'  => __( 'A link to a product sold on a different website, e.g. brand collab.', 'woocommerce' ),
+				'product_template_order'        => 30,
+				'product_template_icon'         => null,
+				'product_template_layout_id'    => 'simple-product',
+				'product_template_product_data' => array(
+					'type' => 'external',
+				),
+			),
+			'variable' => array(
+				'product_template_id'           => 'variable-product-template',
+				'product_template_title'        => __( 'Variable product', 'woocommerce' ),
+				'product_template_description'  => __( 'A product with variations like color or size.', 'woocommerce' ),
+				'product_template_order'        => 40,
+				'product_template_icon'         => null,
+				'product_template_layout_id'    => 'simple-product',
+				'product_template_product_data' => array(
+					'type' => 'variable',
+				),
+			),
+		);
+
+		// Getting the product types registered via the classic editor.
+		// This is not required when registering new product templates.
+		$registered_product_types = wc_get_product_types();
+
+		foreach ( $registered_product_types as $product_type => $title ) {
+			if ( array_key_exists( $product_type, $supported_product_types ) ) continue;
+
+			$supported_product_types[ $product_type ] = array(
+				'product_template_id'           => $product_type . '-product-template',
+				'product_template_title'        => $title,
+				'product_template_description'  => null,
+				'product_template_order'        => ( count( $supported_product_types ) + 1 ) * 10,
+				'product_template_icon'         => null,
+				'product_template_layout_id'    => null,
+				'product_template_product_data' => array(
+					'type' => $product_type,
+				),
+			);
+		}
+
+		foreach ( $supported_product_types as $template_data ) {
+			$product_template = new ProductTemplate(
+				$template_data['product_template_id'],
+				$template_data['product_template_title'],
+				$template_data['product_template_product_data'],
+			);
+			if ( isset( $template_data['product_template_layout_id'] ) ) {
+				$product_template->set_layout_template_id( $template_data['product_template_layout_id'] );
+			}
+			if ( isset( $template_data['product_template_description'] ) ) {
+				$product_template->set_description( $template_data['product_template_description'] );
+			}
+			if ( isset( $template_data['product_template_order'] ) ) {
+				$product_template->set_order( $template_data['product_template_order'] );
+			}
+			if ( isset( $template_data['product_template_icon'] ) ) {
+				$product_template->set_icon( $template_data['product_template_icon'] );
+			}
+
+			$templates[] = $product_template;
+		}
```

Comment:
Why not just create `ProductTemplate` instances initially instead of creating separate data objects and then looping through them?

Reply:
done